osu.Game/Rulesets/Edit/PlacementBlueprint.cs

```diff
@@ -38,16 +37,18 @@ public abstract partial class PlacementBlueprint : CompositeDrawable, IKeyBindin
         /// </summary>
         public readonly HitObject HitObject;
 
-        [Resolved(canBeNull: true)]
-        protected EditorClock EditorClock { get; private set; }
+        [Resolved]
+        protected EditorClock EditorClock { get; private set; } = null!;
 
         [Resolved]
-        private EditorBeatmap beatmap { get; set; }
+        private EditorBeatmap beatmap { get; set; } = null!;
+
+        private Bindable<double> startTimeBindable = null!;
 
-        private Bindable<double> startTimeBindable;
+        private HitObject? getPreviousHitObject() => beatmap.HitObjects.TakeWhile(h => h.StartTime <= startTimeBindable.Value).LastOrDefault();
```

Comment:
Slightly weird that there's one more linq call here now compared to previous version.

```suggestion
        private HitObject? getPreviousHitObject() => beatmap.HitObjects.LastOrDefault(h => h.StartTime <= startTimeBindable.Value);
```

Also I'm noting the equality change (was `<`, is `<=`) but I'd like to think it shouldn't realistically matter...

Reply:
The change to `<=` felt more correct. As for why there's more than one linq call: the version I've provided should be faster as it can early abort at the first failure. The original version would iterate over all items.